src/chat/ChatScreen.js

```diff
@@ -47,128 +46,140 @@ type Props = $ReadOnly<{|
   ...SelectorProps,
 |}>;
 
-type State = {|
-  editMessage: EditMessage | null,
-  fetchError: Error | null,
-|};
-
 const componentStyles = createStyleSheet({
   screen: {
     flex: 1,
     flexDirection: 'column',
   },
 });
 
-class ChatScreen extends PureComponent<Props, State> {
-  static contextType = ThemeContext;
-  context: ThemeData;
-
-  state = {
-    editMessage: null,
-    fetchError: null,
-  };
-
-  // This could live in `this.state`, but it isn't used in `render`.
-  // If we don't use `this.setState` to update it, we don't have to
-  // worry about pointless rerenders. We just check it in
-  // `componentDidUpdate`.
-  shouldFetchWhenNextFocused: boolean = false;
+/**
+ * Fetch messages for this narrow and report an error, if any
+ *
+ * See `MessagesState` for background about the fetching, including
+ * why this is nearly the only place where additional data fetching
+ * is required.  See `fetchMessagesInNarrow` and `fetchMessages` for
+ * more details, including how Redux is kept up-to-date during the
+ * whole process.
+ */
+const useFetchMessages = (args: {|
+  dispatch: Dispatch,
+  isFocused: ?boolean,
+  narrow: Narrow,
+  loading: boolean,
+  fetching: Fetching,
+  eventQueueId: number,
+|}) => {
+  const { dispatch, isFocused, narrow, loading, fetching, eventQueueId } = args;
 
-  componentDidMount() {
-    this.fetch();
-  }
+  const isFetching = fetching.older || fetching.newer || loading;
 
-  /* eslint-disable react/no-did-update-set-state */
-  componentDidUpdate(prevProps, prevState) {
-    const { isFocused } = this.props;
+  // This could live in state, but then we'd risk pointless rerenders;
+  // we only use it in our `useEffect` callbacks. Using `useRef` is
+  // like using instance variables in class components:
+  //   https://reactjs.org/docs/hooks-faq.html#is-there-something-like-instance-variables
+  const shouldFetchWhenNextFocused = React.useRef<boolean>(false);
 
-    // When the event queue changes, schedule a fetch
-    if (prevProps.eventQueueId !== this.props.eventQueueId) {
-      this.shouldFetchWhenNextFocused = true;
-    }
+  const [fetchError, setFetchError] = React.useState<Error | null>(null);
 
-    // Do a scheduled fetch, if it's time
-    if (this.shouldFetchWhenNextFocused && isFocused === true) {
-      this.shouldFetchWhenNextFocused = false;
-      this.fetch();
-    }
-  }
-
-  clearError = () => {
-    this.setState({ fetchError: null });
-  };
-
```

Comment:
While looking closely at this main diff and matching up what bit of old code corresponds to what bit of new code (so as to compare the old and new versions of each bit of code to understand how they've changed), I found that this didn't correspond to anything. And the reason is that it's not actually used!

So another good small prep commit would be to delete this function.